src/plugins/validation/2and3/semantic-validators/schema-ibm.js

```diff
@@ -415,6 +426,57 @@ function checkPropNames(schema, contextPath, config, messages) {
   });
 }
 
+/**
+ * Check that property names follow the specified case convention
+ * @param schema
+ * @param contextPath
+ * @param caseConvention a string, 'off' | 'warning' | 'error'
+ */
+function checkPropNamesCaseCollision(
+  schema,
+  contextPath,
+  caseConvention,
+  messages
+) {
+  if (!schema.properties || !caseConvention) {
+    return;
+  }
+
+  // flag any property whose name does not follow the case convention
+  const prevProps = {};
+  forIn(schema.properties, (property1, propName1) => {
+    prevProps[propName1] = true;
+    forIn(schema.properties, (property2, propName2) => {
+      // Skip explored properties before property1
+      if (prevProps[propName2] == true) return;
+
+      // Skip duplicate properties
+      if (propName1 == propName2) return;
+
+      if (propName1.slice(0, 2) === 'x-') return;
+      if (propName2.slice(0, 2) === 'x-') return;
+
+      // Skip case_convention checks for deprecated properties
+      if (property1.deprecated === true) return;
+      if (property2.deprecated === true) return;
+
+      const checkStatus = caseConvention || 'off';
+      if (checkStatus.match('error|warning')) {
+        const caselessPropName1 = propName1.replace(/[_-]/g, '').toLowerCase();
+        const caselessPropName2 = propName2.replace(/[_-]/g, '').toLowerCase();
+        const isCaseCollision = caselessPropName1 == caselessPropName2;
+        if (isCaseCollision) {
+          messages.addMessage(
+            contextPath.concat(['properties', propName1]),
+            `Property names are identical except for the naming conventions: ${propName1} ~ ${propName2}`,
+            checkStatus
+          );
+        }
+      }
+    });
+  });
```

Comment:
I think we can use a simpler algorithm here. This is a little tough to read. Here is some pseudo-code as a starting point:
```
create an array to hold used names
for name in properties:
  if property shouldn't be processed, skip (deprecated, etc.)
  make name caseless
  if array includes name, error
  else, add name to array
```

I don't think you need to worry about duplicate properties, those should be caught elsewhere. I also think it's okay to only list one of the property names in the error message.